front/lib/resources/data_source_resource.ts

```diff
@@ -272,12 +273,37 @@ export class DataSourceResource extends ResourceWithVault<DataSource> {
     return getDataSourceUsage({ auth, dataSource: this.toJSON() });
   }
 
+  // sId logic.
+
+  get sId(): string {
+    return DataSourceResource.modelIdToSId({
+      id: this.id,
+      workspaceId: this.workspaceId,
+    });
+  }
+
+  static modelIdToSId({
+    id,
+    workspaceId,
+  }: {
+    id: ModelId;
+    workspaceId: ModelId;
+  }): string {
+    return makeSId("data_source", {
+      id,
+      workspaceId,
+    });
+  }
+
+  static isDataSourceSId(sId: string): boolean {
+    return isResourceSId("data_source", sId);
+  }
+
   // Serialization.
 
   toJSON(): DataSourceType {
     return {
       id: this.id,
-      sId: this.name, // TODO(thomas 20240812) Migrate to a real sId
```

Comment:
isn't `sId` missing here ? `sId: this.sId` , it's used in `VaultWebsiteModal`

Reply:
The goal is actually to remove it to explicitely use .name everywhere because it's a name. Then re-introduce it and migrate to using the new .sId everywhere :+1: